src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -141,7 +142,7 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     tests.Add(test);
             }
 
-            return tests.Count > 0
+            return tests.Count > 0 || builders.OfType<TestCaseSourceAttribute>().Any()
```

Comment:
With `builders.Count != 0`, wouldn't that affect theories?
These are all the things that `builders` may contain:
 * **`TheoryAttribute`**
 * `SequentialAttribute`
 * `PairwiseAttribute`
 * `CombinatorialAttribute`
 * `TestCaseSourceAttribute`
 * `TestCaseAttribute`
 * Future implementations of `ITestBuilder`